### PR TITLE
Add bindings to mobile view channel header

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Add bindings to mobile view channel header. Bindings will in the top if there is only one binding/plugin, and in the channel header menu.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34550

#### Related Pull Requests
None

### CODE REVIEW

**Review by hmhealey — APPROVED** [review]

File: plugins/mobile_channel_header_plug/mobile_channel_header_plug.test.jsx [hunk 1]
```diff
@@ -137,13 +149,15 @@ describe('plugins/MobileChannelHeaderPlug', () => {
...
-        const wrapper = mount(
+        const wrapper = mountWithIntl(
             <MobileChannelHeaderPlug
                 components={[newTestPlug]}
                 channel={channel}
                 channelMember={channelMember}
                 theme={{}}
                 isDropdown={true}
+                appsEnabled={false}
+                actions={{doAppCall: jest.fn()}}
```
Comment (mickmister) [inline plugins/mobile_channel_header_plug/mobile_channel_header_plug.test.jsx:160]: Can we add one test where `appsEnabled` is true? Just to make sure `doAppCall` is called on click

File: plugins/mobile_channel_header_plug/mobile_channel_header_plug.tsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+import {Theme} from 'mattermost-redux/types/preferences';
+
+import {PluginComponent} from 'types/store/plugins';
+import {createCallContext, createCallRequest} from 'utils/apps';
+
+type Props = {
+
+    /*
+         * Components or actions to add as channel header buttons
+         */
```
Comment (mickmister) [inline plugins/mobile_channel_header_plug/mobile_channel_header_plug.tsx:22]: The comments here seem to be misaligned compared to the javascript file this came from

File: plugins/mobile_channel_header_plug/mobile_channel_header_plug.tsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    isDropdown: boolean;
+    channel: Channel;
+    channelMember?: ChannelMembership | null;
+
+    /*
+         * Logged in user's theme
+         */
+    theme: Theme;
+    appBindings?: AppBinding[];
```
Comment (mickmister) [inline plugins/mobile_channel_header_plug/mobile_channel_header_plug.tsx:36]: I think this is always defined, based on the `makeAppBindingsSelector` function

File: plugins/mobile_channel_header_plug/mobile_channel_header_plug.tsx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+         * Components or actions to add as channel header buttons
+         */
+    components?: PluginComponent[];
+
+    /*
+         * Set to true if the plug is in the dropdown
+         */
+    isDropdown: boolean;
+    channel: Channel;
+    channelMember?: ChannelMembership | null;
```
Comment (mickmister) [inline plugins/mobile_channel_header_plug/mobile_channel_header_plug.tsx:30]: This prop was marked as required in the js file. Should it be required here?

**Review by mickmister — CHANGES_REQUESTED** [review]
Great job fixing this @larkox. I have one request to add a test to ensure `doAppCall` is called when a binding button is clicked.

File: plugins/mobile_channel_header_plug/mobile_channel_header_plug.tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+        const callResp = (res as {data: AppCallResponse}).data;
+        const ephemeral = (message: string) => sendEphemeralPost(message, this.props.channel.id);
+        switch (callResp.type) {
+        case AppCallResponseTypes.OK:
+            if (callResp.markdown) {
+                ephemeral(callResp.markdown);
+            }
+            break;
+        case AppCallResponseTypes.ERROR: {
+            const errorMessage = callResp.error || this.props.intl.formatMessage({id: 'apps.error.unknown', defaultMessage: 'Unknown error happenned'});
```
Comment (catalintomai) [inline plugins/mobile_channel_header_plug/mobile_channel_header_plug.tsx:153]: s/happenned/happened

**Review by catalintomai — APPROVED** [review]

File: plugins/mobile_channel_header_plug/mobile_channel_header_plug.tsx (hunk 4 above)
Comment (larkox) [reply plugins/mobile_channel_header_plug/mobile_channel_header_plug.tsx:30]: After further inspection, this is better to leave as `channelMember?: ChannelMembership`. GetCurrentChannelMembership was returning null, and only was being checked as a falsey value, so we can just leave it as undefined.

**Review by mickmister — APPROVED** [review]
LGTM, great job on the tests :+1:

**Comment by DHaussermann** [comment]
Sorry for the delay in getting this approved. 

@amyblais please cheery-pick if it's possible.

**Pull request merged**